### ISSUE TITLE
Expiration date on share doesn't do anything!?

### ISSUE THREAD

**Issue by user897943** [photoview/photoview#666]
**Describe the bug**
I typed a date on the expiration date field of the share settings for an album, I tried USA format mm/dd/yyyy and UK format dd/mm/yyyy but neither has worked. There is no button to save the setting? I believe a date picker on that field would be helpful to avoid confusion with date system (US vs rest of world). I also think the button to save the setting should be present like it is for the password field.

**To Reproduce**
1. Go to 'gear next to album settings'
2. Click on 'add share, then edit the share, try adding a date'
3. Nothing happens!

**Expected behavior**
There is a date picker, and the expiration date gets saved.

**Screenshots**
not applicable

**Your environment:**
Feel free to remove and add what you think might be useful.

 - Are you using Docker? yes
 - Database mariadb latest as of march 25 2022
 - Browser edge, firefox
 - Version 2.3.12 (0b9d04ef)

**Additional context**

**Comment by viktorstrate** [comment]
This has never been implemented on the UI side

**Comment by user897943** [comment]
You're wrong. It most definitely is on the UI!

![Screenshot_20220325-223455](https://user-images.githubusercontent.com/5891221/160210839-366e58f0-f779-4215-88d4-ec6933dfd231.png)

**Comment by viktorstrate** [comment]
Yes, it's only a mock-up. It was never wired up.

**Comment by user897943** [comment]
> Yes, it's only a mock-up. It was never wired up.

That's different than saying "it was never in the UI", if it's not working it shouldn't be exposed, the whole reason I'm here is because it's confusing, if anyone sees a setting, they assume it does something, shouldn't be in the released version if it's not "wired up"...

**Comment by viktorstrate** [comment]
I agree with you, I'm just trying to explain the reason why it doesn't work. Sorry for the confusion.

**Comment by user897943** [comment]
> I agree with you, I'm just trying to explain the reason why it doesn't work. Sorry for the confusion.

Ah okay, makes sense!!

**Comment by vik-s** [comment]
I just ran into this myself, and I'm glad I ran into the explanation on this github issue. Maybe a good way to implement this would be to have the suggestion in #384 to have different time frames like 1 day, 1 week, etc, and finally a custom date that can be picked from a calendar view. The sharing works great, and I use it all the time. Hooking up the expiration date to do its thing would be awesome! Thanks for all the work @viktorstrate!!

**Comment by nathanrdodson** [comment]
I will be working on this @viktorstrate - is there already expiry functionality in the back-end, as in if the expiry was set and passed by, the share would be inaccessible? Or does the API need to built out for this entirely?

I see an `expire` directive in the `share_token.go`, but am unsure if it functions if set.
